include/cell/copy/global_to_shared.hpp

```diff
@@ -0,0 +1,170 @@
+#pragma once
+
+#include "cell/copy/constants.hpp"
+#include "cell/copy/dyn_copy.hpp"
+#include "cell/copy/warp.hpp"
+#include "cell/traits/base.hpp"
+#include "types/mod.hpp"
+
+namespace tiledcuda::cell::copy {
+
+using namespace traits;
+namespace tl = tile_layout;
+
+template <typename Global_, typename Shared_, const int kThreads_,
+          const tl::Layout kType, typename Base>
+struct GlobalToSharedLoaderImpl {
+    using Global = Global_;
+    using Shared = Shared_;
+    using DType = typename Global::DType;
+
+    DEVICE void operator()(const DType* src, DType* dst);
+};
+
+template <typename Global_, typename Shared_, const int kThreads_,
+          typename Base>
+struct GlobalToSharedLoaderImpl<Global_, Shared_, kThreads_,
+                                tl::Layout::kSwizzledRowMajor, Base> {
+    using Global = Global_;
+    using Shared = Shared_;
+    using DType = typename Global::DType;
+
+    static constexpr int kThreads = kThreads_;
+
+    static constexpr int kRows = Global::kRows;
+    static constexpr int kCols = Global::kCols;
+    static constexpr int kShmRows = Shared::kRows;
+    static constexpr int kShmCols = Shared::kCols;
+
+    // To avoid bank conflict, the shared memory requires a swizzled layout
+    static const int kSwizzleMode = kShmCols % 32 ? 1 : 0;
+    using Swizzled =
+        tl::SwizzledRowMajor<DType, kShmRows, kShmCols, kSwizzleMode>;
+
+    using SrcLayout = tl::RowMajor<kRows, kCols, kCols>;
+    using DstLayout = typename Swizzled::SmemLayout;
+
+    // threads in a thread block are laid out as a 2D tile
+    // that has a shape of kThreadsRows x kThreadsCols.
+    static constexpr int kThreadsCols = kShmCols / Base::kNumPerAccess;
+    static constexpr int kThreadsRows = kThreads / kThreadsCols;
```

Comment:
If we consider lines 17 and 49-50 together, there seems to be an inconsistency. We allow users to define a `WarpLayout` to arrange threads within a thread block, but then lines 49-50 ignore the user's settings and compute an internal thread layout instead.

This behavior is confusing because the internal system does not work in the manner described by the exposed interface.

Reply:
This is indeed a problem worth considering, as in order to maintain consistency in the `Load/Store` struct format, we have used the `WaroLayout` as a template parameter for passing. At the same time, we have calculated the actual number of threads used by the user based on the `WarpLayout`. However, for **Swizzle**, the internal organization of the threads is performed once again, and contrary to the user's expectation, this may not run according to the original user's organization of the Warp. To some extent, the `WarpLayout` has been discarded.

Reply:
I noticed some inconsistencies. If an API is exposed to external users, its internal behavior should align with the interfaces it provides. Otherwise, it could cause issues for potential automated analysis.

Let's work together to resolve this issue in other pull requests.